core/src/main/scala/org/apache/spark/ui/scope/RDDOperationGraph.scala

```diff
@@ -221,6 +226,24 @@ private[spark] object RDDOperationGraph extends Logging {
     RDDOperationGraph(internalEdges.toSeq, outgoingEdges.toSeq, incomingEdges.toSeq, rootCluster)
   }
 
+  def getRepeatIdentifier(node: RDDOperationNode): String = {
```

Comment:
We can add one control point here with a new configuration so that users can enable and disable this implementation at any time.

Reply:
I don't see how we can add that easily in the code given that RDDOperationGraph doesn't have a SqlConf instance to read a config.